PathPlanning/DynamicWindowApproach/dynamic_window_approach.py

```diff
@@ -101,7 +100,6 @@ def calc_dynamic_window(x, config):
           x[4] - config.max_delta_yaw_rate * config.dt,
           x[4] + config.max_delta_yaw_rate * config.dt]
 
-    #  [v_min, v_max, yaw_rate_min, yaw_rate_max]
```

Comment:
Could you please put back the comment? I think it is useful.

Reply:
Sorry for the missing comment, I have made some changes in this part before for the brake velocity limit, however when testing the code, I found the robot keep moves in a very low speed thus delete this change, below is the explanation for the idea:
in order to brake in front of an obstacle, the max speed limitation should not greater than sqrt(2*distance*acceleration), therefore I bring this into consideration when generate dynamic window.
The basic idea is find the distance to the nearest obstacle, then calculate the brake speed limit and compare with the Vd and Vs in the original code to find the dynamic window.
This changes handle the situations that the original code failed, which is in some obstacle maps (the obstacle map and goal I have provided before), the original code drive the circular shape robot just across through the obstacles because the velocity is too high to brake (while the rectangle robots works fine).
However, this change also bring a problem which is I don’t know how to filter the below situation in a simple manner: 

![demo](https://user-images.githubusercontent.com/21169260/92263769-c1037a80-ef0f-11ea-9f91-e99a774053f3.PNG)

As can be seen from figure, in this situation, the obstacle is not in the way of robot, so we should not consider the brake speed when generate dynamic window, the brake velocity should only be considered when the obstacle is just in front of the robot heading.
My changes in brake velocity part didn’t filter this situation, therefore although it can handle some problems, the speed is significantly limited since the robot is continueously assume it need to brake in front of the nearest obstacle (which actually may not in its way), this makes the robot moves very slow.

Under this consideration, I delete the brake velocity part and modified the to_goal_cost_gain=0.2 and obstacle_cost_gain=2.0, this helps to solve the robot cross obstacle problems and meanwhile keep the robot move in a higher speed.